# CMU/Philips Capstone Progress
## Dec 9th

## Loading in files
Wrote code for extracting relevant info from most of the given documents. Given a filepath (or files selected in UI), should be able to load in files successfully.

In [1]:
from read_write import *

### Reading CO Matrix

In [4]:
# Trace matrix (can process CO and PSC separately)
trace_matrix = read_trace("..\data\ER 2228015 v43 ATT1 Sapphire 1.33 Trace Matrix.xlsx", "CO")
trace_matrix

Loading CO from: ..\data\ER 2228015 v43 ATT1 Sapphire 1.33 Trace Matrix.xlsx
Loading in worksheet titled: Care Orchestrator PRD / SRS Trace Matrix


PRD   SRS ID     Method  \
0        US3528   TC1000  Automated   
1        US3528   TC1000     Manual   
2        US2978   TC1001     Manual   
3       US18461  TC10011     Manual   
4       US18461  TC10011     Manual   
...         ...      ...        ...   
123738   US3528    TC995  Automated   
123739   US3528    TC996     Manual   
123740   US3528    TC996     Manual   
123741   US3528    TC996     Manual   
123742  US32334   TC9995     Manual   

                                                Test Name V&V Test Report  
0          TC1000 tests.downloads.POU.tcCompliancePOUTest     2228014 v51  
1       PRD US3528 US1683 SRS TC996 TC1000 TC2058 View...     2228014 v29  
2       PRD US2978 SRS TC1001 The patient activity log...     2228014 v02  
3       PRD US32334 SRS TC11148 TC9673 TC9757 TC10011 ...     2228014 v45  
4       PRD US32334 SRS TC11148 TC9679 TC9758 TC10011 ...     2228014 v45  
...                                                   ...             ...  
123738  TC995 TC2924 TC2931 TC2932 tests.task.complian...     2228014 v51  
123739  PRD US3528 US1683 SRS TC996 TC1000 TC2058 View...     2228014 v29  
123740  PRD US3528 US1683 SRS TC996 TC1998 TC1999 Comp...     2228014 v33  
123741  PRD US3528 US1683 SRS TC996 TC1998 TC1999 Comp...     2228014 v33  
123742  PRD US32334 SRS TC11148 TC9682 TC9995 TC10011 ...     2228014 v45  

[123743 rows x 5 columns]

We've manually verified that 123743 rows are in the original trace file.

<div>
<img src="demo_resources/co_bottom.png" width=50%>
</div>

(2 empty rows and headers on top)

<div>
<img src="demo_resources/co_top.png" width=50%>
</div>

### Reading PSC Matrix

In [6]:
psc_trace = read_trace("..\data\ER 2228015 v43 ATT1 Sapphire 1.33 Trace Matrix.xlsx", "PSC")
psc_trace

Loading PSC from: ..\data\ER 2228015 v43 ATT1 Sapphire 1.33 Trace Matrix.xlsx
Loading in worksheet titled: Care Orchestrator Product Support Console PRD / SRS Trace Matrix


PRD   SRS ID     Method  \
0      US32341  TC10051  Automated   
1      US32341  TC10051  Automated   
2      US32341  TC10051  Automated   
3      US32341  TC10051  Automated   
4      US32341  TC10051  Automated   
...        ...      ...        ...   
68619  US18461   TC9759  Automated   
68620  US35549   TC9779     Manual   
68621   US9787   TC9845     Manual   
68622  US35549   TC9956  Automated   
68623  US35549   TC9957  Automated   

                                               Test Name V&V Test Report  
0                                TC10051 Document Upload     2228014 v51  
1                           TC10051 patient list display     2228014 v51  
2                         TC10051 patient record display     2228014 v51  
3                         TC10051 patient search display     2228014 v51  
4                                 TC10051 Task List view     2228014 v51  
...                                                  ...             ...  
68619  TC9759 tests.task.prescription.PrescriptionTas...     2228014 v51  
68620    PRD US13044 SRS TC9779 Verify filter and search     2228014 v46  
68621  PRD US9787 SRS TC9845 Deletion of LTAM reminde...     2228014 v41  
68622  TC9299 TC9956 tests.CIS.CISV1Tests.CISCreateNe...     2228014 v51  
68623  TC9300 TC9957 tests.CIS.CISV1Tests.CISGetEnabl...     2228014 v51  

[68624 rows x 5 columns]

### Reading Automatic RestApiTests
There's a lot of text files here that are stored in nested folders.
<div>
<img src="demo_resources/api_tests.png" width=30%>
</div>

Instead of selecting each of these files, you can select the main RestApiTests folder and the program can read in all of the relevant sub-files. 

In [1]:
from read_write import *

In [2]:
api_tests = read_rest_api_tests("../data/RestApiTests")
api_tests

Loading api test files from ../data/RestApiTests


test_name  status   rc  \
0      TC6281 com.philips.sapphire.systemintegrationt...  passed  RC1   
1      TC6281 com.philips.sapphire.systemintegrationt...  passed  RC1   
2      TC5871 com.philips.sapphire.systemintegrationt...  passed  RC1   
3      TC6282 com.philips.sapphire.systemintegrationt...  passed  RC1   
4      TC5867 TC5868 TC5869 TC5872 TC5873 TC5874 TC58...  passed  RC1   
...                                                  ...     ...  ...   
12739  TC460 TC464 TC1126 TC1189 TC2149 com.philips.s...  passed  RC4   
12740  TC460 TC464 TC1126 TC1189 TC2148 com.philips.s...  passed  RC4   
12741  TC460 TC464 TC1126 TC1189 TC2146 com.philips.s...  passed  RC4   
12742  TC237 TC2069 TC2542 com.philips.sapphire.syste...  passed  RC4   
12743  TC2331 com.philips.sapphire.systemintegrationt...  passed  RC4   

           name                                          file_name  
0      Akshitha  ..\data\RestApiTests\RC1\Akshitha\20200924T110...  
1      Akshitha  ..\data\RestApiTests\RC1\Akshitha\20200924T110...  
2      Akshitha  ..\data\RestApiTests\RC1\Akshitha\20200924T110...  
3      Akshitha  ..\data\RestApiTests\RC1\Akshitha\20200924T110...  
4      Akshitha  ..\data\RestApiTests\RC1\Akshitha\20200924T110...  
...         ...                                                ...  
12739    Mardis  ..\data\RestApiTests\RC4\Mardis\Mardis\2020100...  
12740    Mardis  ..\data\RestApiTests\RC4\Mardis\Mardis\2020100...  
12741    Mardis  ..\data\RestApiTests\RC4\Mardis\Mardis\2020100...  
12742    Mardis  ..\data\RestApiTests\RC4\Mardis\Mardis\2020100...  
12743    Mardis  ..\data\RestApiTests\RC4\Mardis\Mardis\2020100...  

[12744 rows x 5 columns]

In [7]:
# Unique names (folder names) in the folders
print(f"Names for each folder:\n{list(api_tests.name.unique())}")

Names for each folder:
['Akshitha', 'Carlos', 'Fischer', 'Gulfam', 'Mardis', 'Subhash', 'TeamCity', 'Edison', 'Jeff', 'Keefe', 'Landy', 'Mani', 'Nihar', 'Gazzo']


In [ ]:
# Unique RC #s
print(f"Names for each folder:\n{list(api_tests.name.unique())}")

In [4]:
# Number of unique file names
print(f"# of unique api test .txt files: {len(api_tests.file_name.unique())}")
api_tests.file_name.unique()

# of unique api test .txt files: 508


hod.txt',
       '..\\data\\RestApiTests\\RC3\\Jeff\\Jeff\\20201006T170022Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Jeff\\Jeff\\20201006T171157Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Jeff\\Jeff\\20201007T174021Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Keefe\\20201001T192928Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Keefe\\20201001T194727Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Keefe\\20201006T180047Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Keefe\\Keefe\\20201001T192928Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Keefe\\Keefe\\20201001T194727Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Keefe\\Keefe\\20201006T180047Z-LegacyTrace-GroupByMethod.txt',
       '..\\data\\RestApiTests\\RC3\\Landy\\20201006T200803Z\\20201006T200803Z-LegacyTrace-GroupByMethod.txt',
      

In [5]:
# Compare to an independent file count
from pathlib import Path

api_tests_folder = Path("../data/RestApiTests")
len([str(f) for f in api_tests_folder.rglob("*.txt") if "__MACOSX" not in str(f)])

508